File: frontend/src/types/label.ts
export interface LabelResult {
  id: string;
  passed: boolean;
  errors: string[];
  warnings: string[];
  extracted_data: {
    brand_name: string | null;
    class_type: string | null;
    alcohol_content: string | null;
    net_contents: string | null;
    bottler_address: string | null;
    country_of_origin: string | null;
    government_warning: string | null;
  };
  processing_time_ms: number;
}

File: backend/app/models/label.py
from pydantic import BaseModel
from typing import Optional, List

class LabelData(BaseModel):
    brand_name: Optional[str] = None
    class_type: Optional[str] = None
    alcohol_content: Optional[str] = None
    net_contents: Optional[str] = None
    bottler_address: Optional[str] = None
    country_of_origin: Optional[str] = None
    government_warning: Optional[str] = None

class ValidationResult(BaseModel):
    passed: bool
    errors: List[str]
    warnings: List[str]
    extracted_data: LabelData